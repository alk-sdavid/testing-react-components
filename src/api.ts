interface OptionsType {
  [key: string]: boolean;
}

export async function fetchOptions(): Promise<OptionsType> {
  await new Promise((resolve) => setTimeout(resolve, 500));
  return { a: true, b: false, c: false };
}

export async function saveOptions(options: OptionsType): Promise<OptionsType> {
  await new Promise((resolve) => setTimeout(resolve, 500));
  return options;
}
