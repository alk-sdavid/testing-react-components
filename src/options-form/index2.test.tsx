import { render, waitFor } from '@testing-library/react';
import { OptionsForm } from '.';
import { fetchOptions } from '../api';

jest.mock('../api', () => ({
  fetchOptions: jest.fn(() => Promise.resolve({ a: true })),
}));

describe('<OptionsForm />', () => {
  afterEach(() => {
    (fetchOptions as jest.Mock).mockClear();
  });

  it('should be well rendered', async () => {
    const { container } = render(<OptionsForm sendNotification={() => {}} />);
    await waitFor(() => expect(fetchOptions).toHaveBeenCalled());
    expect(container).toMatchSnapshot();
  });

  it('should be still well rendered', async () => {
    const { container } = render(
      <OptionsForm sendNotification={() => {}} canUpdate={false} />,
    );
    await waitFor(() => expect(fetchOptions).toHaveBeenCalled());
    expect(container).toMatchSnapshot();
  });
});
