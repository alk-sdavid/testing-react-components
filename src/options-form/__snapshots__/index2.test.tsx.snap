// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionsForm /> should be still well rendered 1`] = `
<div>
  <ul>
    <li>
      <input
        checked=""
        id="option-a"
        readonly=""
        type="checkbox"
      />
      <label
        for="option-a"
      >
        Option 
        A
      </label>
    </li>
  </ul>
  <button
    disabled=""
  >
    Save
  </button>
</div>
`;

exports[`<OptionsForm /> should be well rendered 1`] = `
<div>
  <ul>
    <li>
      <input
        checked=""
        id="option-a"
        readonly=""
        type="checkbox"
      />
      <label
        for="option-a"
      >
        Option 
        A
      </label>
    </li>
  </ul>
  <button
    disabled=""
  >
    Save
  </button>
</div>
`;
